packages/cli/src/cog/cog.stac.job.ts

```diff
@@ -0,0 +1,416 @@
+                rel: 'self',
+            },
+            {
+                href: 'job.json',
+                type: 'application/json',
+                rel: 'job',
+            },
+        ];
+        let sourceStac = {} as StacCollection;
+
+        const interval: [string, string][] = [];
+        try {
+            const sourceCollectionPath = FileOperator.join(ctx.sourceLocation.path, 'collection.json');
+            sourceStac = await FileOperator.readJson<StacCollection>(sourceCollectionPath);
+            description = sourceStac.description;
+            interval.push(...(sourceStac.extent?.temporal?.interval ?? []));
+            links.push({ href: sourceCollectionPath, rel: 'sourceImagery', type: 'application/json' });
+            if (sourceStac.providers != null) providers.push(...sourceStac.providers);
+        } catch (err) {
+            if (!FileOperator.isCompositeError(err) || err.code !== 404) {
+                throw err;
+            }
+        }
+        const keywords = sourceStac.keywords ?? CogStacKeywords.slice();
+        const license = sourceStac.license ?? CogStacLicense;
+        const title = sourceStac.title ?? titleizeImageryName(imageryName);
+
+        const job = new CogStacJob({
+            id,
+            name: imageryName,
+            title,
+            description,
+            source: {
+                gsd: metadata.pixelScale,
+                epsg: metadata.projection,
+                files: metadata.bounds,
+                location: ctx.sourceLocation,
+            },
+            output: {
+                gsd: ctx.targetProjection.tms.pixelScale(metadata.resZoom),
+                epsg: ctx.targetProjection.tms.projection.code,
+                files: metadata.files,
+                location: ctx.outputLocation,
+
+                resampling: ctx.override?.resampling ?? GdalCogBuilderDefaults.resampling,
+                quality: ctx.override?.quality ?? GdalCogBuilderDefaults.quality,
+                cutline: ctx.cutline,
+                addAlpha,
+                nodata: metadata.nodata,
+                bounds: metadata.targetBounds,
+                oneCogCovering: ctx.oneCogCovering,
+            },
+        });
+
+        const nowStr = new Date().toISOString();
+
+        const sourceProj = job.sourcePtms.proj;
+
+        const bbox = sourceProj.boundsToWgs84BoundingBox(
+            metadata.bounds.map((a) => Bounds.fromJson(a)).reduce((sum, a) => sum.union(a)),
+        );
+
+        if (interval.length == 0) {
+            const years = extractYearRangeFromName(imageryName);
+            if (years[0] != -1) {
+                interval.push(years.map((y) => `${y}-01-01T00:00:00Z`) as [string, string]);
+            }
+        }
+
+        if (ctx.cutline) {
+            links.push({ href: 'cutline.geojson.gz', type: 'application/geo+json+gzip', rel: 'cutline' });
+        }
+
+        links.push({ href: 'covering.geojson', type: 'application/geo+json', rel: 'covering' });
```

Comment:
same comments with `rel` if we are making them up do we want to namespace our ones?